tests/backup/backup_upgrade_test.go

```diff
@@ -759,3 +760,552 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", func() {
+			dash.VerifyFatal(dstClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", DestinationClusterName))
+			destClusterUid, err = Inst().Backup.GetClusterUID(ctx, BackupOrgID, DestinationClusterName)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster uid", DestinationClusterName))
+			log.Infof("Cluster [%s] uid: [%s]", DestinationClusterName, destClusterUid)
+		})
+		Step("Create pre and post exec rules for applications", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			preRuleNames = make(map[string]string, 0)
+			preRuleUids = make(map[string]string, 0)
+			log.InfoD("Creating pre exec rules for applications %v", Inst().AppList)
+			for _, appName := range Inst().AppList {
+				log.Infof("Creating pre backup rule for application [%s]", appName)
+				_, preRuleName, err := Inst().Backup.CreateRuleForBackup(appName, BackupOrgID, "pre")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of pre backup rule for application [%s]", appName))
+				preRuleUid := ""
+				if preRuleName != "" {
+					preRuleUid, err = Inst().Backup.GetRuleUid(BackupOrgID, ctx, preRuleName)
+					log.FailOnError(err, "Fetching pre backup rule [%s] uid", preRuleName)
+					log.Infof("Pre backup rule [%s] uid: [%s]", preRuleName, preRuleUid)
+				}
+				for i := 0; i < len(srcClusterAppNamespaces[appName]); i++ {
+					preRuleNames[appName] = preRuleName
+					preRuleUids[appName] = preRuleUid
+				}
+			}
+			postRuleNames = make(map[string]string, 0)
+			postRuleUids = make(map[string]string, 0)
+			log.InfoD("Creating post exec rules for applications %v", Inst().AppList)
+			for _, appName := range Inst().AppList {
+				log.Infof("Creating post backup rule for application [%s]", appName)
+				_, postRuleName, err := Inst().Backup.CreateRuleForBackup(appName, BackupOrgID, "post")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of post-backup rule for application [%s]", appName))
+				postRuleUid := ""
+				if postRuleName != "" {
+					postRuleUid, err = Inst().Backup.GetRuleUid(BackupOrgID, ctx, postRuleName)
+					log.FailOnError(err, "Fetching post backup rule [%s] uid", postRuleName)
+					log.Infof("Post backup rule [%s] uid: [%s]", postRuleName, postRuleUid)
+				}
+				for i := 0; i < len(srcClusterAppNamespaces[appName]); i++ {
+					postRuleNames[appName] = postRuleName
+					postRuleUids[appName] = postRuleUid
+				}
+			}
+		})
+		Step("Create backups with and without pre and post exec rules", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			backupToContextMapping = make(map[string][]*scheduler.Context, 0)
+			createBackupWithRulesTask := func(appName string) {
+				namespace := srcClusterAppNamespaces[appName][0]
+				backupName := fmt.Sprintf("%s-%s-%v-with-rules", BackupNamePrefix, namespace, time.Now().Unix())
```

Comment:
How about RandomString(4) instead of time.Now().Unix() since we are running in parallel?

Reply:
I would prefer to have both it helps in finding out the sequence of the backups.